xpallets/assets/src/lib.rs

```diff
@@ -24,210 +24,131 @@ pub mod weights;
+        type TreasuryAccount: TreasuryAccount<Self::AccountId>;
+
+        /// The hook for doing something on the event of creating an account.
+        type OnCreatedAccount: HandleLifetime<Self::AccountId>;
+
+        /// The hook triggered whenever the asset balance of an account is changed.
+        type OnAssetChanged: OnAssetChanged<Self::AccountId, BalanceOf<Self>>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// transfer between two accounts
+        #[pallet::weight(<T as Config>::WeightInfo::zero())]
         pub fn transfer(
-            origin,
+            origin: OriginFor<T>,
             dest: <T::Lookup as StaticLookup>::Source,
-            #[compact] id: AssetId,
-            #[compact] value: BalanceOf<T>
+            #[pallet::compact] id: AssetId,
+            #[pallet::compact] value: BalanceOf<T>,
         ) -> DispatchResult {
             let transactor = ensure_signed(origin)?;
             let dest = T::Lookup::lookup(dest)?;
             debug!(target: "runtime::assets", "[transfer] from:{:?}, to:{:?}, id:{}, value:{:?}", transactor, dest, id, value);
             Self::can_transfer(&id)?;
 
-            Self::move_usable_balance(&id, &transactor, &dest, value).map_err::<Error::<T>, _>(Into::into)?;
+            Self::move_usable_balance(&id, &transactor, &dest, value)
+                .map_err::<Error<T>, _>(Into::into)?;
 
             Ok(())
         }
 
-        /// for transfer by root
-        #[weight = 0]
+        /// transfer method reserved for root(sudo)
+        #[pallet::weight(<T as Config>::WeightInfo::zero())]
         pub fn force_transfer(
-            origin,
+            origin: OriginFor<T>,
             transactor: <T::Lookup as StaticLookup>::Source,
             dest: <T::Lookup as StaticLookup>::Source,
-            #[compact] id: AssetId,
-            #[compact] value: BalanceOf<T>
+            #[pallet::compact] id: AssetId,
+            #[pallet::compact] value: BalanceOf<T>,
         ) -> DispatchResult {
             ensure_root(origin)?;
 
             let transactor = T::Lookup::lookup(transactor)?;
             let dest = T::Lookup::lookup(dest)?;
             debug!(target: "runtime::assets", "[force_transfer] from:{:?}, to:{:?}, id:{}, value:{:?}", transactor, dest, id, value);
             Self::can_transfer(&id)?;
-            Self::move_usable_balance(&id, &transactor, &dest, value).map_err::<Error::<T>, _>(Into::into)?;
+            Self::move_usable_balance(&id, &transactor, &dest, value)
+                .map_err::<Error<T>, _>(Into::into)?;
             Ok(())
         }
 
         /// set free token for an account
-        #[weight = 0]
+        #[pallet::weight(<T as Config>::WeightInfo::zero())]
```

Comment:
```suggestion
        #[pallet::weight(0)]
```
Why not use 0 directly as before?